snakemake/scheduler.py

```diff
@@ -263,7 +263,23 @@ def schedule(self):
                         job.reset_params_and_resources()
 
                     logger.debug(f"Resources before job selection: {self.resources}")
-                    logger.debug(f"Ready jobs: {len(needrun)}")
+
+                    # Subsample jobs to be run (to speedup solver)
+                    n_total_needrun = len(needrun)
+                    solver_max_jobs = int(
+                        os.environ.get("SNAKEMAKE_SOLVER_MAX_JOBS", sys.maxsize)
```

Comment:
Let us rather have this as a command line parameter. Then people can also set it via profiles.